src/stores/grid_renderer_store.ts

```diff
@@ -465,26 +452,40 @@ export class GridRenderer {
 
     // column text + separator
     for (const i of visibleCols) {
-      const colSize = this.getters.getColSize(sheetId, i);
       const colName = numberToLetters(i);
       ctx.fillStyle = activeCols.has(i) ? "#fff" : TEXT_HEADER_COLOR;
-      let colStart = this.getHeaderOffset("COL", left, i);
+      const zone = { left: i, right: i, top: top, bottom: top };
+      const { x: colStart, width: colSize } = this.getters.getRect(zone);
+      const { x, width } = this.getters.getVisibleRect(zone);
+      ctx.save();
+      ctx.beginPath();
```

Comment:
There's also a `ctx.beginPath();` before the for-loop that might not be necessary anymore (?)